miio/airpurifier_miot.py

```diff
@@ -80,6 +82,37 @@
     "favorite_rpm": {"siid": 9, "piid": 3},
 }
 
+# https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:air-purifier:0000A007:zhimi-va2:2
+_MODEL_AIRPURIFIER_VA2 = {
```

Comment:
Please start converting the `_MAPPING` to be a dict of mappings, so that we can avoid having multiple different interfaces in the future. This should allow dropping the separate classes for both AirpurifierMiot as well as for the status container, while giving us a common interface among models.

Reply:
If all models use a common interface, how to set or get a model-specific property ?

Reply:
The interface can be extended case-by-case basis. So for getters (so mostly related to status container), the way is to return `None` if the value is not available. And for setters, the model can be checked (`self.model`) to raise an exception on non-supported devices.

Reply:
I met some problems during coverting the `_MAPPING` to be a dict of mappings.
Here's the dict of mappings
```python
_MODEL_MAPPING = {
    "zhimi.airpurifier.ma4": _MAPPING,  # airpurifier 3
    "zhimi.airpurifier.mb3": _MAPPING,  # airpurifier 3h
    "zhimi.airpurifier.va1": _MAPPING,  # airpurifier proh
    "zhimi.airpurifier.vb2": _MAPPING,  # airpurifier proh
    "zhimi.airpurifier.mb4": _MAPPING_MB4,  # airpurifier 3c
    "zhimi.airp.mb4a": _MAPPING_MB4,  # airpurifier 3c
    "zhimi.airp.va2": _MAPPING_VA2,  # airpurifier 4 pro
}
```
However, how to specify the mapping in `AirPurifierMiot` class ?
self.model is not valid here, and self.model is an optional parameter even if I specify the mapping in a `__init__` function
```python
class AirPurifierMiot(BasicAirPurifierMiot):
    """Main class representing the air purifier which uses MIoT protocol."""

    mapping = _MODEL_MAPPING[self.model]
    _supported_models = _MODEL_MAPPING.keys()
```

Reply:
Since #1302, you just need to define `_mappings` and all relevant methods use the correct mapping in the background, see https://github.com/rytilahti/python-miio/blob/master/miio/heater_miot.py#L141 for an example.